dbt_loom/__init__.py

```diff
@@ -36,13 +38,19 @@ class DbtCloudReferenceConfig(BaseModel):
     api_endpoint: Optional[str] = None
     step: Optional[int] = None
 
+class GCSReferenceConfig(BaseModel):
+    """Configuration for a GCS reference"""
+    
+    bucket_name: str
+    blob_name: str
+    credentials: Optional[str] = None
```

Comment:
I'd recommend being explicit with the type for `credentials`. Since `from_service_account_json` is expecting a file path, let's use a `Path` type from `pathlib` instead of `str`. This has the added bonus letting us perform normalization and expansion of the path. This will allow users to provide both relative and absolute paths to the config file.